pkg/controllers/applicationset_controller.go

```diff
@@ -265,11 +267,17 @@ func getTempApplication(applicationSetTemplate argoprojiov1alpha1.ApplicationSet
 	return &tmplApplication
 }
 
+func mergeGeneratorTemplate(g generators.Generator, requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, applicationSetTemplate argoprojiov1alpha1.ApplicationSetTemplate) argoprojiov1alpha1.ApplicationSetTemplate {
+	dest := g.GetTemplate(requestedGenerator)
```

Comment:
How does this change look:
```go
func mergeGeneratorTemplate(g generators.Generator, requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, applicationSetTemplate argoprojiov1alpha1.ApplicationSetTemplate) (argoprojiov1alpha1.ApplicationSetTemplate, error) {

	// Make a copy of the value from `GetTemplate()` before merge, rather than copying directly into
	// the provided parameter (which will touch the original resource object returned by client-go)
	dest := g.GetTemplate(requestedGenerator).DeepCopy()

	err := mergo.Merge(dest, applicationSetTemplate)

	return *dest, err
}
```

This: 
- Makes a deep copy of the `requestedGenerator` parameter before merging, so we don't accidentally modify it with the `Merge` call
- Return the error from `Merge`

Sound good?

Reply:
thanks